tests/test_kb_dataset.py

```diff
@@ -0,0 +1,310 @@
+import argparse
+import sys
+import unittest
+from collections import defaultdict
+from difflib import ndiff
+from pathlib import Path
+from pprint import pformat
+from typing import Iterator, Optional, Union
+
+sys.path.append(str(Path(__file__).parent.parent))
+
+import datasets
+
+from schemas.kb import features
+
+OFFSET_ERROR_MSG = (
+    "\n"
+    "There are features with wrong offsets!"
+    "This is not a hard failure, as it is common for this type of datasets"
+    "However, if the error list is long (e.g. >10) you should double check your code"
+)
+
+
+def _get_example_text(example: dict) -> str:
+    """
+    Get full text example
+    """
+    return " ".join([t for p in example["passages"] for t in p["text"]])
+
+
+def _check_offsets(
+    example_text: str,
+    offsets: list[list[int]],
+    texts: list[str],
+) -> Iterator:
+
+    for idx, (start, end) in enumerate(offsets):
+
+        try:
+            text = texts[idx]
+
+        # TODO: riase error?
+        except IndexError:
+            print("Pairing of a list of offsets with a single text is not allowed!")
```

Comment:
I think we finally landed on treating all offsets and text the same way in the schema so we can probably pull this out to its own test that asserts, 
* all offsets are of the form `[(lo1, hi1), ...]`
* all text fields that are paired with offsets are of the form `[text, ..]`
* offsets in passages have only one element `[(lo1,hi1)]`
* texts in passages have only one element `[text]`

no strong feelings about raising error or how its presented to the user, just saying we can treat all offsets the same.

Reply:
My opinion is that adding tests for this would be helpful. This way we ensure the format. 

When I wrote this I was thinking more of the situation of an entity which has multiple offsets, e.g. should the format be

"offsets" : [(lo1, hi1), (lo1, hi1),...] , "text" : ["text"]

or do we ask to duplicate the text?

"offsets" : [(lo1, hi1), (lo1, hi1),...] , "text" : ["text", "text"]

Reply:
> len(offsets) == len(text)
Is this true? What about entities with same text on multiple locations?